internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -918,9 +915,9 @@ func (r *AgentServiceConfigReconciler) newImageServiceConfigMap(ctx context.Cont
 	return cm, mutateFn, nil
 }
 
-func (r *AgentServiceConfigReconciler) urlForRoute(ctx context.Context, routeName string) (string, error) {
+func (r *AgentServiceConfigReconciler) urlForRoute(ctx context.Context, routeName string, instance *aiv1beta1.AgentServiceConfig) (string, error) {
```

Comment:
If namespace is the only thing you need for this function, why not just pass that instead of the whole instance?

Reply:
I am passing an instance reference because it is more clear that way. On the called function then it is obvious what namespace is being used.